### ISSUE TITLE
Guard against possible race condition in new CNI

### ISSUE THREAD

**Issue by slonka** [kumahq/kuma#4560]
### Description

TLDR:

> If an application pod starts before CNI installer installs CNI, Kubelet has no knowledge of the CNI plugin. The result is that the application pod comes up without iptables rules.

The race condition is best described in the [Istio design doc](https://docs.google.com/document/d/1SQzrFxtcn3o_79OtJYbSHMPuqZNhR3EaEhbkpBVMXAw/edit#bookmark=id.u0d2bzyzcq8x)

Istio first solved the issue by labeling the broken pods with a specific label and then killing them ([PR](https://bit.ly/first_approach_pr)). However this approach causes additional churning during a node start. It also produces a number of errors in customer logs and lacks portability. So they switched to the taint controller approach implemented in this [PR](https://bit.ly/taint_controller_pr).

Possible solutions:
1. Reimplement the taint controller
2. At DP startup check if transparent proxy is set and fail the start if it's not.
3. Try to revive "[KEP: Node Readiness Gates #1003](https://bit.ly/kep_node_readiness)"

WDYT?

**Comment by slonka** [comment]
I started implementing this and it seems that we have a default probe that will not go healthy if the traffic is not redirected to Envoy. This prevents workloads from being accessed so if we couple that with having the CNI on startup getting a list of pods and injecting the rules inside those pods then we should be fine against this race condition. WDYT?
